src/main/resources/filing/templates/small-full-accounts.html

```diff
@@ -0,0 +1,1503 @@
+              <ix:nonNumeric contextRef="Director1_CY" name="core:DirectorSigningFinancialStatements" format="ixt2:nocontent"></ix:nonNumeric>
+              <ix:nonNumeric name="uk-bus:EndDateForPeriodCoveredByReport" contextRef="CY_END">{{$current_period_end_on}}</ix:nonNumeric>
+              <ix:nonNumeric name="uk-bus:BalanceSheetDate" contextRef="CY_END">{{$current_period_end_on}}</ix:nonNumeric>
+              
+              <ix:nonNumeric contextRef="CY_END" name="uk-bus:StartDateForPeriodCoveredByReport">{{$current_period_start_on}}</ix:nonNumeric>
+
+              <ix:nonNumeric contextRef="CY" name="uk-bus:DescriptionPrincipalActivities">No description of principal activity</ix:nonNumeric>
+              <ix:nonNumeric contextRef="CY" name="uk-bus:EntityDormantTruefalse" format="ixt2:booleanfalse">false</ix:nonNumeric>
+              <ix:nonNumeric contextRef="CY" name="uk-bus:EntityTradingStatus" format="ixt2:nocontent"></ix:nonNumeric>
+            
+            </ix:hidden>
+            <ix:references>
+              <link:schemaRef xlink:type="simple" xlink:href="https://xbrl.frc.org.uk/FRS-102/2014-09-01/FRS-102-2014-09-01.xsd" />
+            </ix:references>
+            <ix:resources>
+              <xbrli:unit id="GBP">
+                <xbrli:measure>iso4217:GBP</xbrli:measure>
+              </xbrli:unit>
+              <xbrli:unit id="pure">
+                <xbrli:measure>xbrli:pure</xbrli:measure>
+              </xbrli:unit>
+              {{if $previous_period_start_on}}
+                <xbrli:context id="PY">
+                  <xbrli:entity>
+                    <xbrli:identifier scheme="http://www.companieshouse.gov.uk/">{{$companyNumber}}</xbrli:identifier>
+                  </xbrli:entity>
+                  <xbrli:period>
+                      <xbrli:startDate>{{$previous_period_start_on}}</xbrli:startDate>
+                      <xbrli:endDate>{{$previous_period_end_on}}</xbrli:endDate>
+                  </xbrli:period>
+                </xbrli:context>
+              {{end}} 
+              <xbrli:context id="CY_END">
+                <xbrli:entity>
+                  <xbrli:identifier scheme="http://www.companieshouse.gov.uk/">{{$companyNumber}}</xbrli:identifier>
+                </xbrli:entity>
+                <xbrli:period>
+                  <xbrli:instant>{{$current_period_end_on}}</xbrli:instant>
+                </xbrli:period>
+              </xbrli:context>
+              <xbrli:context id="CY">
+                <xbrli:entity>
+                  <xbrli:identifier scheme="http://www.companieshouse.gov.uk/">{{$companyNumber}}</xbrli:identifier>
+                </xbrli:entity>
+                <xbrli:period>
+                  <xbrli:startDate>{{$current_period_start_on}}</xbrli:startDate>
+                  <xbrli:endDate>{{$current_period_end_on}}</xbrli:endDate>
+                </xbrli:period>
+              </xbrli:context>
+              {{if $previous_period_start_on}}
```

Comment:
Is it sufficient to check that `previous_period_start_on` exists to determine whether to print `previous_period_end_on`? 

We'd hope the data was good enough that both exist if `previous_period_start_on` exists, but what if it isn't? 

Just food for thought, but maybe we could pass `previous_period_exists` in the model which would be set logically in the service. I'm guessing we use this condition a lot over the balance sheet.

That being said, this logic is used everywhere in the abridged template at the moment anyway.